nwbinspector/checks/tables.py

```diff
@@ -74,6 +75,36 @@ def check_time_intervals_stop_after_start(time_intervals: TimeIntervals, nelems:
         )
 
 
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_column_binary_capability(table: DynamicTable, nelems: int = 200):
+    """
+    Check each column of a table to see if the data could be set as a boolean dtype.
+
+    Parameters
+    ----------
+    time_intervals: DynamicTable
+    nelems: int
+        Only check the first {nelems} elements. This is useful in case there columns are
+        very long so you don't need to load the entire array into memory. Use None to
+        load the entire arrays.
+    """
+    for column in table.columns:
+        if hasattr(column, "data") and not isinstance(column, VectorIndex):
+            if np.asarray(column.data[0]).itemsize == 1:
+                return  # already boolean, int8, or uint8
```

Comment:
```suggestion
                yield # already boolean, int8, or uint8
```

I think if this is `return` it won't proceed to examine any other columns on that table.

Reply:
ah